src/reducetensor.cpp

```diff
@@ -1031,6 +1032,10 @@ void ReduceTensorDescriptor::ReduceTensor(const Handle& handle,
                                        std::to_string(static_cast<int>(use_padding.first)) +
                                        std::to_string(static_cast<int>(use_padding.second));
 
+        int alignment            = 64;
+        Data_t workspace_aligned = std::align(
+            alignment, workspaceSizeInBytes - alignment, workspace, workspaceSizeInBytes);
```

Comment:
It doesn't look like we need to subtract alignment here: `workspaceSizeInBytes - alignment`, `std::align` does it automatically and store the resulted size into the last parameter. But it does not break anything and works fine right now.

Reply:
The two parameters `size` and  `space` of `std::align` are confusing and seems no use here.  My codes here refer to the one used in `src/ocl/rnnocl.cpp`

Reply:
> It doesn't look like we need to subtract alignment here: `workspaceSizeInBytes - alignment`, `std::align` does it automatically and store the resulted size into the last parameter. But it does not break anything and works fine right now.

Using the same value in the 2nd and last arg would be an incorrect way to use std:align.
The second argument of the function is the size of the structure after alignment. The fourth argument is the size of the buffer in which the aligned pointer should be located. If the buffer size is insufficient for alignment, the function returns nullptr and does not change the pointer or workspaceSizeInBytes.

workspaceSizeInBytes = real_size + (space for alignment(64));
therefore, it is necessary to send only the actual size of the aligned structure to the second argument.